it/uk/gov/hmrc/nrs/retrieval/NrsRetrievalIntegrationSpec.scala

```diff
@@ -0,0 +1,217 @@
+
+  "GET /nrs-retrieval/retrieval/ping" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/retrieval/ping").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "GET /nrs-retrieval/submission-metadata" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/submission-metadata").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "GET /nrs-retrieval/submission-bundles/:vaultId/:archiveId" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/submission-bundles/vaultId/archiveId").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "HEAD /nrs-retrieval/submission-bundles/:vaultId/:archiveId" should {
+    val url = s"$serviceRoot/submission-bundles/vaultId/archiveId"
+    val requestWithoutHeaders = wsClient.url(url)
+    val requestWithHeader = wsClientWithXApiKeyHeader(url)
+
+    def givenSubmissionBundlesReturns(status: Int, withHeaders: Boolean): Unit =
+      wireMockServer.addStubMapping(
+        mappingBuilder(
+          head(urlPathMatching("/retrieval/submission-bundles/vaultId/archiveId")), withHeaders
+        ).willReturn(aResponse().withStatus(status)).build())
+
+    Seq(
+      (requestWithHeader, true, "the X-API-Key header"),
+      (requestWithoutHeaders, false, "no headers")
+    ).foreach{ case (request, withHeaders, headersLabel) =>
+      "pass through the X-API-Key header if set and return OK" when {
+        s"the request has $headersLabel and the submission bundles request returns OK" in {
+          givenSubmissionBundlesReturns(OK, withHeaders)
+          request.head().futureValue.status shouldBe OK
+        }
+
+       s"the request has $headersLabel and the submission bundles request returns a 2xx status" in {
+          givenSubmissionBundlesReturns(CREATED, withHeaders)
+          request.head().futureValue.status shouldBe OK
+        }
+      }
+
+      "pass through the X-API-Key header if set and return ACCEPTED" when {
+        s"the request has $headersLabel and the submission bundles request returns ACCEPTED" in {
+          givenSubmissionBundlesReturns(ACCEPTED, withHeaders)
+          request.head().futureValue.status shouldBe ACCEPTED
+        }
+      }
+
+      "pass through the X-API-Key header if set and return NOT_FOUND" when {
+        s"the request has $headersLabel and the submit request returns NOT_FOUND" in {
+          givenSubmissionBundlesReturns(NOT_FOUND, withHeaders)
+          request.head().futureValue.status shouldBe NOT_FOUND
+        }
+      }
+
+      "pass through the X-API-Key header if set and return INTERNAL_SERVER_ERROR" when {
+        s"the request has $headersLabel and the submit request returns a 3xx status" in {
+          givenSubmissionBundlesReturns(SEE_OTHER, withHeaders)
+          request.head().futureValue.status shouldBe INTERNAL_SERVER_ERROR
+        }
+
+        Seq(BAD_REQUEST, UNAUTHORIZED, FORBIDDEN).foreach{ status =>
+          s"the request has $headersLabel and the submit request returns a 4xx status $status" in {
```

Comment:
also here 4xx status should be INTERNAL_SERVER_ERROR

Reply:
confusing, I didn't add comments in top to bottom order

Reply:
It's basically documenting the behaviour of `http-verbs`

It is confusing in that it is not transparent.

OTOH it kind of makes sense. If our service passes a bad request to another service, and so our service receives a 4xx response, then from a client perspective this is an INTERNAL_SERVICE_ERROR. That client hasn't made a BAD_REQUEST - the clients request has gone past our validation and been accepted, but our service is unable to process it as it makes a bad request elsewhere.

Equally if we get a 5xx response back form another service, then this is a BAD_GATEWAY error from a client perspective. 

So from that perspective the non-transparent behaviour makes sense.